src/service/viewport/viewport-impl.js

```diff
@@ -425,20 +429,27 @@ export class ViewportImpl {
 
   /** @override */
   animateScrollIntoView(element, pos = 'top', opt_duration, opt_curve) {
-    devAssert(
-      !opt_curve || opt_duration !== undefined,
-      "Curve without duration doesn't make sense."
-    );
+    if (IS_SXG) {
+      const posToBlock = {'top': 'start', 'center': 'center', 'bottom': 'end'};
+      return tryResolve(() =>
```

Comment:
Not sure if this is the correct thing to do here, however there is a mechanism that determines duration in viewport that I could use. @dvoytenko

Reply:
What @erwinmombay suggested is a combination of both, so I went with that.